src/v2/api/resources/ResourceByIdPipe.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable, PipeTransform } from '@nestjs/common';
+import { Subject } from '@prisma/client';
+import { InvalidEntityIdException } from '../../utils/exceptions/InvalidEntityIdException';
+import { ResourceService } from './ResourcesService';
+
+@Injectable()
+export class ResourceByIdPipe implements PipeTransform<string, Promise<string>> {
+  constructor(
+    private resourceService: ResourceService
+  ) {}
+
+  async transform(value: string): Promise<string> {
+    const resource: Subject = await this.resourceService.get(value);
```

Comment:
it's better to use ResourceRepository for such purpose. Also, remove unused types (Subject, Promise<string>, and \<string, Promise\<string\>\>)